flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -411,30 +413,31 @@ private static Request createRequest(
             String targetUrl,
             HttpMethod httpMethod,
             ByteBuf jsonPayload,
-            Collection<FileUpload> fileUploads)
+            Collection<FileUpload> fileUploads,
+            Collection<HttpHeader> customHeaders)
             throws IOException {
         if (fileUploads.isEmpty()) {
 
             HttpRequest httpRequest =
                     new DefaultFullHttpRequest(
                             HttpVersion.HTTP_1_1, httpMethod, targetUrl, jsonPayload);
 
-            httpRequest
-                    .headers()
-                    .set(HttpHeaderNames.HOST, targetAddress)
+            HttpHeaders headers = httpRequest.headers();
+            headers.set(HttpHeaderNames.HOST, targetAddress)
                     .set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE)
                     .add(HttpHeaderNames.CONTENT_LENGTH, jsonPayload.capacity())
                     .add(HttpHeaderNames.CONTENT_TYPE, RestConstants.REST_CONTENT_TYPE);
+            customHeaders.forEach(ch -> headers.add(ch.getName(), ch.getValue()));
 
             return new SimpleRequest(httpRequest);
         } else {
             HttpRequest httpRequest =
                     new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, httpMethod, targetUrl);
 
-            httpRequest
-                    .headers()
-                    .set(HttpHeaderNames.HOST, targetAddress)
+            HttpHeaders headers = httpRequest.headers();
+            headers.set(HttpHeaderNames.HOST, targetAddress)
                     .set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+            customHeaders.forEach(ch -> headers.set(ch.getName(), ch.getValue()));
```

Comment:
nit: any reason to use `headers.set(...)` here, and `headers.add(...)` on [line 430](https://github.com/apache/flink/pull/22816/files#diff-d6cdab55fefade687bd62fb086b6fa85aa4f8c91a745e8ec406fb8ac8982b9e1R430) ?
If no specific reason, using the same method might make it more readable.

Reply:
addressed